spec/request/api/registration/registrations_spec.rb

```diff
@@ -0,0 +1,122 @@
+      expect(response_json["status"]).to eq "success"
+    end
+  end
+
+  describe "Unsuccessfully register - with wrong password" do
+    before do
+      post "/api/auth",
+           params: {
+             email: "client@mail.com",
+             password: "password",
+             password_confirmation: "wrongpassword",
+             role: "client",
+             company_name: "company name",
+             company_url: "http://develup.com",
+           },
+           headers: headers
+    end
+
+    it "is expected to respond with unprocessable entity status" do
+      expect(response).to have_http_status :unprocessable_entity
+    end
+
+    it "returns a unsuccesfully message" do
+      expect(response_json["errors"]["full_messages"][0]).to eq "Password confirmation doesn't match Password"
+    end
+  end
+
+  describe "Unsuccessfully register - with invalid email" do
+    before do
+      post "/api/auth",
+           params: {
+             email: "clien tttttt@mail",
+             password: "password",
+             password_confirmation: "password",
+             role: "client",
+             company_name: "company name",
+             company_url: "http://develup.com",
+           },
+           headers: headers
+    end
+
+    it "is expected to respond with unprocessable entity status" do
+      expect(response).to have_http_status :unprocessable_entity
+    end
+
+    it "returns a unsuccesfully message" do
+      expect(response_json["errors"]["email"][0]).to eq "is not an email"
+    end
+  end
+
+  describe "Unsuccesfuly register - email already exists" do
+    let!(:user) { create(:user, email: "client@mail.com") }
+    before do
+      post "/api/auth",
+           params: {
+             email: "client@mail.com",
+             password: "password",
+             password_confirmation: "password",
+             role: "client",
+             company_name: "company name",
+             company_url: "http://develup.com",
+           },
+           headers: headers
+    end
+
+    it "is expected to respond with unprocessable entity status" do
+      expect(response).to have_http_status :unprocessable_entity
+    end
+
+    it "returns a unsuccesfully message" do
+      expect(response_json["errors"]["full_messages"][0]).to eq "Email has already been taken"
+    end
+  end
+
+  describe "Unsuccesfuly register -with blank company_name, company_url and role" do
+    before do
+      post "/api/auth",
+           params: {
+             email: "client@mail.com",
+             password: "password",
+             password_confirmation: "password",
+           },
+           headers: headers
+    end
+
+    it "is expected to respond with unprocessable entity status" do
+      expect(response).to have_http_status :unprocessable_entity
+    end
+
+    it "returns a unsuccesfully message if role is blank" do
+      expect(response_json["errors"]["full_messages"][0]).to eq "Role can't be blank"
+    end
```

Comment:
role shouldnt return a errormessage as the user shouldnt fill in this one?

Reply:
The front should send that data. If fore some reason that doens't happen you won't be able to create a user and you will receive an error message

Reply:
aah, get it! Thanks!